### ISSUE TITLE
[BUG] r25c does not seem to be able to produce l.so that are compatable with source debugging.

### ISSUE THREAD

**Issue by ScatteredRay** [android/ndk#1874]
### Description

When compiling with ndk r25c, and launching an application on device, and attempting to debug with correct symbol locations, we are not able to get source level debugging with android studio.

in the LLDB window, running an image lookup such as:

```image lookup -vn MyGame::Initialize```

The CompileUnit will not be correct on MS Windows, it looks like:

``` CompileUnit: id = {0x0000036a}, file = "/Workspace/Project/Source/MyEngine.cpp", language = "c++11"```

However, doing the same on r23c I get what I expect which is:

``` CompileUnit: id = {0x0000036a}, file = "D:\Workspace\Project\Source\MyEngine.cpp", language = "c++11"```

and source level debugging works correctly.

Oddly enough, using the llvm.symbolizer.exe provided in the ndk r25c is able to find the correct source path and print source context just find.

### Affected versions

r25

### Host OS

Windows

### Host OS version

Windows 10

### Affected ABIs

arm64-v8a

### Build system

ndk-build

### Other build system

CMake, Ninja, NDK-Build, Gradle

### minSdkVersion

28

**Comment by DanAlbert** [comment]
> in the LLDB window

> Oddly enough, using the llvm.symbolizer.exe provided in the ndk r25c is able to find the correct source path and print source context just find.

This smells like an Android Studio issue. @emrekultursay in case it sounds familiar, but to me it doesn't look like there's enough info for us to proceed. About the only time we don't need a repro case is when it's a duplicate.

**Comment by ScatteredRay** [comment]
I'd love to provide a repro-case, but sadly I work for a company who's games are proprietary and would have to be subject to an NDA. I'd try to construct one manually, but I'm really not an android engineer and putting together an android project from scratch may be beyond my capability.

FWIW, this is with Android studio 2022.1.1.21, and is happening on all of our games on all branches under every SDK version we're trying to build against. So I'd be pretty surprised if it was unique to our game. And based on our prior experience with the Android SDK suspect it's likely broken functionality with that particular version of the components.

**Comment by DanAlbert** [comment]
> And based on our prior experience with the Android SDK suspect it's likely broken functionality with that particular version of the components.

I don't at all doubt that :) The question is whether it's this component or something else. The fact that llvm-symbolizer works makes me think it may be LLDB failing to interpret the debug info produced by LLVM rather than LLVM producing bad debug info. Without a binary we can inspect I can't prove anything one way or the other. @emrekultursay may be able to tell you what info he needs short of a full repro case.

**Comment by ScatteredRay** [comment]
So, as for a reproduction, in this case, Android Studio seems to be able to better handle the descrepency, but I'm able to reproduce a symptom with ndk-examples:

1.  download and extract https://dl.google.com/android/repository/android-ndk-r23c-windows.zip and https://redirector.gvt1.com/edgedl/android/studio/ide-zips/2022.2.1.18/android-studio-2022.2.1.18-windows.zip
2. setup environment: to include ndk and android studio (likely contains some unused environment:
```
ANDROID_NDK=C:\packages\AndroidNDK.25.2.0
ANDROID_NDK_ROOT=C:\packages\AndroidNDK.25.2.0
ANDROID_NDK_TOOLCHAINS_PATH=C:\packages\AndroidNDK.25.2.0\toolchains
JAVA_HOME=C:\packages\AndroidStudio-2022.2.1.18\jbr
ANDROID_HOME=C:\Android\
PATH+=C:\packages\AndroidNDK.25.2.0\;C:\packages\AndroidStudio-2022.2.1.18\jbr\bin;C:\packages\AndroidStudio-2022.2.1.18\bin
```
3. download and build hello-vulkan from ndk-samples
```
C:
cd \dev\
git-clone https://github.com/android/ndk-samples.gitc
cd .\ndk-samples\
cd .\hello-vulkan\
.\gradlew assembleDebug
```
4. launch AndroidStudio included via PATH `studio64`
5. Browse to hello-vulkan project
6. launch project for debugging, and wait for fault or break while running.
7. in LLDB window run ```image lookup -vn HelloVK::render```
8. Notice that the CompileUnit is 
```
CompileUnit: id = {0x00000000}, file = "/dev/ndk-samples/hello-vulkan/app/.cxx/Debug/6d3c7020/arm64-v8a/D:/dev/ndk-samples/hello-vulkan/app/src/main/cpp/vk_main.cpp", language = "c++14"
```

**Comment by ScatteredRay** [comment]
So, I was able to get the same project to build in NDK 23 (Mostly by changing `ndkVersion '23.2.8568313'` in `hello-vulkan/app/build.gradle`

And the output is `CompileUnit: id = {0x00000000}, file = "D:\dev\ndk-samples\hello-vulkan\app\src\main\cpp\vk_main.cpp", language = "c++14"

Now, not 100% sure this is the problem, since AndroidStudio is still able to debug both versions, but it's highly suspect to me...

**Comment by emrekultursay** [comment]
The following patch fixed that issue in LLDB: https://reviews.llvm.org/D122389

An LLDB that includes that was shipped with Android Studio 2022.2.1 Flamingo . Older versions of Android Studio (including Android Studio 2022.1.1 Electric Eel) does not have that patch. Please try Flamingo and let us know if it works for you.

Android NDK also ships a command-line version of LLDB (at `toolchains\llvm\prebuilt\windows-x86_64\bin\lldb[.exe, .cmd]`) , and the LLDB that is shipped with NDK r25 contains the above patch.
